.github/workflows/release.yml

```diff
@@ -0,0 +1,23 @@
+name: Release
+on:
+  push:
+    tags:
+      - "*"
+jobs:
+  create-release-note:
+    name: Create release note
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 3.1
+      - name: Extract release note
+        run: |
+          ruby -e 'print("## "); puts(ARGF.read.split(/^## /)[1])' CHANGELOG.md \
+            > release-note.md
+      - name: Upload to release
+        uses: softprops/action-gh-release@v1
```

Comment:
mmm. Is this 3rd-party action safe?

I do understand the convenience, but what happens if `softprops/action-gh-release` got compromised?

Reply:
If this got compromised and wrong releases were created or no releases were created, we can stop using this and continue to create releases manually as before (if we still need releases).

Reply:
If that's the case, I can see the point that we don't have to use this in the first place.

Reply:
I think it is fine not to use this and not to create a release as before,
or to create releases manually without this.

But if the latter is the case, I think we can use this as long as it works without any problems.

Reply:
@daipom I should have explained it clearer. Here is my understanding:

* GitHub Actions has `write` privilege to the repository.
  * For example, we can run "git commit && git push" in the action.
  * This is because the execution environment contains a valid `GITHUB_TOKEN`.
* In this sense, we implicitly grant broad privileges to the 3rd-party plugin.
* The impacts can vary from "embarassing" to "pretty serious".
  * e.g. "Hey, look at this funny release note posted on [fluent-plugin-out-http/releaases](https://github.com/fluent-plugins-nursery/fluent-plugin-out-http/releases)."
  * e.g. "Our plugin code was silently compromised with malcious code being embedded"

I think this is why GitHub strongly recommend to be careful of 3rd-party actions:

> This means that a compromise of a single action within a workflow can be
> very significant, as that compromised action would have access to all secrets
> configured on your repository,
>
> https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions

My general feeling is not to include `release.yml` in main. I do like convenience,
but do not think it's worth the potential security risk.

Reply:
> I think it is fine not to use this and not to create a release as before,
> or to create releases manually without this.

Addendum: I'm fine with either way. I don' t think GitHub release is
something we must to do.

Reply:
Oh, I see! Thanks so much for the explanation.
I didn't understand what `compromise` means. It means security risks!